lib/Transforms/ElementwiseToAffine/ElementwiseToAffine.cpp

```diff
@@ -130,10 +129,29 @@ struct ElementwiseToAffine
     RewritePatternSet patterns(context);
 
     patterns.add<ConvertAnyElementwiseMappableOpOnRankedTensors>(context);
-    target.markUnknownOpDynamicallyLegal([](Operation *op) {
+    target.markUnknownOpDynamicallyLegal([&](Operation *op) {
       // TODO (#768): to make this pass more widely applicable, it should take
       // a dialect and/or list of operations to restrict the conversion to
-      return !isElementwiseMappableOpOnRankedTensors(op);
+
+      auto isDialectInList =
+          llvm::any_of(restrictDialects, [&](std::string dialectName) {
+            return dialectName == op->getDialect()->getNamespace().str();
+          });
+      auto isOpInList = llvm::any_of(restrictOps, [&](std::string opName) {
+        return opName == op->getName().getStringRef().str();
```

Comment:
You might be able to simplify these a bit by using `llvm::is_contained` (See [this upstream example of a similar pass option](https://github.com/AlexanderViand-Intel/llvm-project/blob/ebeaebf7f961f8d8c1043e0a5d3dc33758acfdb4/mlir/lib/Dialect/Bufferization/Transforms/Bufferize.cpp#L256-L259))